### PR TITLE
[BD-46] feat: implement Stack component

### PR DESCRIPTION
**Pull request by monteri**
New component `Stack`:
- vertical and horizontal directions
- gap based on `the spacer classes`
- styles are took from the React-Bootstrap `Stack`

JIRA: [PAR-406](https://openedx.atlassian.net/browse/PAR-406)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @monteri! I've created [BLENDED-1100](https://openedx.atlassian.net/browse/BLENDED-1100) to keep track of it in Jira.   More details are on the [BD-46](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3134685276/BD-46+Paragon+Blended+Support+and+Maintenance) project page.


When this pull request is ready, tag your edX technical lead.

File: src/Stack/Stack.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+.pgn__hstack {
+  display: flex;
+  align-self: stretch;
+  gap: $stack-gap;
+
+  @each $level, $space in $spacers {
+    &.pgn__stack-gap--#{$level} {
+      gap: $space;
+    }
+  }
```
Comment (adamstankiewicz) [inline src/Stack/Stack.scss:13]: Nice! 💯

**Review by adamstankiewicz — APPROVED** [review]
LGTM! :shipit:

**Comment by edx-webhook** [comment]
@monteri 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1098&entry.1484655318=https://github.com/monteri&entry.752974735=2022-02-17+11:26&entry.1917517419=2022-02-18+13:26&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**